internal/services/compute/shared_image_resource.go

```diff
@@ -644,3 +443,242 @@ func expandGalleryImageRecommended(d *pluginsdk.ResourceData) (*compute.Recommen
 
 	return result, nil
 }
+
+func expandSharedImageFeatures(d *pluginsdk.ResourceData) *[]compute.GalleryImageFeature {
+	var features []compute.GalleryImageFeature
+	if d.Get("accelerated_network_support_enabled").(bool) {
+		features = append(features, compute.GalleryImageFeature{
+			Name:  utils.String("IsAcceleratedNetworkSupported"),
+			Value: utils.String("true"),
+		})
+	}
+
+	if v, ok := d.GetOk("security_type"); ok {
+		features = append(features, compute.GalleryImageFeature{
+			Name:  utils.String("SecurityType"),
+			Value: utils.String(v.(string)),
```

Comment:
similar to SignalR with it's `features` block, doing this means that it's not trivial to validate these values are set one way or the other with Policy as Code tooling - as such we likely want to instead add booleans for the other options (`ConfidentialVmSupported` and `ConfidentialVM`), rather than introducing a generic field here? That'd allow tooling such as Sentinel/OPA to validate that these are enabled/disabled